### ISSUE TITLE
fix: Federated plugin: inter-package-dependencies should be version-based

### ISSUE THREAD

**Issue by sebastianhaberey** [VeryGoodOpenSource/very_good_cli#178]
**Description**
As I understand it, the main advantage of the federated plugin structure is uncoupling of packages so they can be spread across different repositories and maintained independently. They must also be published as separate plugins. This does not seem possible with the generated project because the packages are linked via file system.

Again, if this is the intended design, feel free to close this (I'm just learning Flutter).

**Steps To Reproduce**

1. `very_good create --org-name com.example --template flutter_plugin foo`
2. File foo/foo_android/pubspec.yaml contains:

```
dependencies:
  foo_platform_interface:
    path: ../foo_platform_interface
```

**Expected Behavior**

```
dependencies:
  foo_platform_interface: ^0.0.1
```

**Comment by felangel** [comment]
Hi @sebastianhaberey 👋 
Thanks for opening an issue!

The reason for the current approach is because when you're creating a new plugin, your platform_interface will likely not already exist on pub.dev so you'll probably want to reference it via path. Eventually once the interface it published, the can be changed but I'd rather leave it as is because if we default to a version like `foo_platform_interface: ^0.0.1`, then the initial `flutter packages get` will fail because it won't be able to find `foo_platform_interface`.

Hope that helps and let me know what you think, thanks!

**Comment by sebastianhaberey** [comment]
Yes you're right, I can see how you'd want to have a newly created project run out-of-the-box. Also it's probably a good practice to start out with file-based references until the project is ready for the initial publish. So I'm closing this issue - thanks for the clarification!

I'm thinking It's an aspect that could be improved about federated plugins. In Java / Maven / IntelliJ, packages are resolved locally first (before the Maven repository). So if I had `foo_platform_interface` 0.0.1 as a local package, IntelliJ would find it and use it. I guess that's what we need for Flutter / Android Studio.
